Campaign/resources/assets/js/bannerSelector.js

```diff
@@ -0,0 +1,848 @@
+import { select } from 'optimal-select';
+
+(function(window, document, undefined) {
+  // initialize variables for elements that are about to be injected
+  var hoveredOverlayElement,
+    hoveredOverlayNameElement,
+    selectedOverlayElement,
+    selectedOverlayNameElement,
+    selectedMenuElement,
+    clearSelectionElement,
+    generatedSelectorWrapperElement,
+    selectorInputElement,
+    generatedSelectorElement,
+    matchingNumberElement,
+    injectBannerElement,
+    bannerInsertionSettingsModalElement,
+    bannerInsertionSettingsModalCancelElement,
+    bannerInsertionSettingsModalAcceptElement,
+    bannerInsertionSettingsSelectElement,
+    cancelAndReturnToRempElement,
+    saveAndReturnToRempElement;
+
+  // window reference from which this site was open
+  var rempParentWebsite;
+  // TODO: move this to some sort of config file, or accept it during init?
+  var allowedOrigin = 'http://campaign.remp.press';
+
+  // markup for injected toolbar
+  var toolbarMarkup = `
```

Comment:
Btw. seeing this as a markup and style in JS looks very like this should have been a Vue component. Nothing to address, just for a future reference.